src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,296 @@
+        final int BY_AT_INDEX = 3;
+        final String TODO_CODE = "T";
+        final String DEADLINE_CODE = "D";
+        final String EVENT_CODE = "E";
+        String line;
+        File dir = new File("data");
+
+        if (!dir.exists()) {
+            dir.mkdir();
+        }
+
+        File f = new File("data/duke.txt");
+        Scanner s = new Scanner(f);
+
+        while (s.hasNext()) {
+            line = s.nextLine();
+            String[] lineData = line.split(LOAD_DELIMITER);
+
+            if (lineData[TASK_INDEX].equals(TODO_CODE)) {
+                userInputs.add(new Todo(lineData[DESCRIPTION_INDEX]));
+            } else if (lineData[TASK_INDEX].equals(DEADLINE_CODE)) {
+                userInputs.add(new Deadline(lineData[DESCRIPTION_INDEX], lineData[BY_AT_INDEX]));
+            } else if (lineData[TASK_INDEX].equals(EVENT_CODE)) {
+                userInputs.add(new Event(lineData[DESCRIPTION_INDEX], lineData[BY_AT_INDEX]));
+            }
+
+            if (lineData[DONE_INDEX].equals("1")) {
+                userInputs.get(userInputsCount).markAsDone();
+            }
+            userInputsCount++;
+        }
+    }
+
+    /**
+     * Prompt for user inputs
+     */
+    public static void promptInput() {
+        String line;
+        Scanner in = new Scanner(System.in);
+        line = in.nextLine();
+
+        while (!line.equals("bye")) {
+            processInput(line);
+            line = in.nextLine();
+        }
+
+        try {
+            printEnd();
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e);
+        }
+    }
+
+    /**
+     * Process user input
+     * @param input the user input
+     */
+    public static void processInput(String input) {
+        input = input.trim();
+
+        try {
+            if (input.startsWith("list")) {
+                listTasks(input);
+            } else if (input.startsWith("done")) {
+                markDone(input);
+            } else if (input.startsWith("todo")) {
+                addTodo(input);
+            } else if (input.startsWith("deadline")) {
+                addDeadline(input);
+            } else if (input.startsWith("event")) {
+                addEvent(input);
+            } else if (input.startsWith("delete")) {
+                deleteTask(input);
+            } else {
+                handleInvalid();
+            }
+        } catch (DukeException e) {
+            System.out.println(e);
+            System.out.println(SEPARATOR);
+        }
+    }
+
+    /**
+     * Prints the ending messages before ending the program
+     */
+    public static void printEnd() throws IOException{
+        final String DELIMITER = "--";
+        int doneValue;
+        FileWriter beginWrite = new FileWriter("data/duke.txt");
+
+        beginWrite.write("");
+        beginWrite.close();
+
+        FileWriter fw = new FileWriter("data/duke.txt", true);
+
+        for (int i = 0; i < userInputsCount; i++) {
```

Comment:
The code within this loop could be improved by [Avoiding Complicated Expressions](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions). You may also consider using `isinstance`.